task/progress.go

```diff
@@ -57,6 +62,11 @@ func calcProgress(count, size uint64) (val float64) {
 	return
 }
 
+func scaleProgress(progress, startFraction, endFraction float64) (val float64) {
+	val = startFraction + progress*(endFraction-startFraction)
+	return
+}
```

Comment:
I think the named return value is only making it more complex here (in fact caused the last bug?). Wdyt of?

```suggestion
func scaleProgress(progress, startFraction, endFraction float64) float64 {
	return startFraction + progress*(endFraction-startFraction)
}
```